docs/K8s-test-plan.md

```diff
@@ -0,0 +1,362 @@
+- Disable flag is set to true (config not yet saved)
+- No manifests applied
+#### Test Steps
+1. Save config (`disable=true` is saved)
+   - **Expect** Kube server status remains `connected=false`
+2. Set `disable=false`
+   - **Expect:** Kube server status updates to `connected=true`
+3. Do config reload
+   - **Expect:** Kube server status updates to `connected=false`, `disable=true` from saved config
+
+### TC_JOIN_5: Config Reload Toggles Disable to False
+#### Test Objective
+Verify Device Under Test (DUT) appropriately disconnects upon config reload toggling disable flag to false. Disable was set to true without being saved prior to config reload. 
+#### Test Configuration
+- Pick up from step 6 of [TC_JOIN_1](#tc_join_1-join-master-once-available)
+- Kube server status shows `connected=true` 
+- Disable flag is set to false (config saved)
+- No manifests applied
+#### Test Steps
+1. Set `disable=true`
+   - **Expect:** Kube server status shows `connected=false`
+2. Do config reload
+   - **Expect:** Kube server status shows `connected=true` , `disable=false` from saved config
+
+
+## Test Scenario: Mode Transitions for {feature x}, Feature Image Upgrades with Reachable Master
+
+### TC_LOCAL_KUBE_1: Switch between Local Mode and Kube Mode
+#### Test Objective
+Verify Device Under Test (DUT) properly transitions between local mode and kube mode when manifest is properly applied.
+#### Test Configuration
+- Pick up from step 6 of [TC_JOIN_1](#tc_join_1-join-master-once-available)
+- {feature x} `current_owner=local`, `set_owner=kube`, `remote_state=none`, `container_version=1.0.0`
+#### Test Steps
+1. Apply manifest for {feature x} v1.1.1 with valid URL
+   - local {feature x} v1.0.0 container should terminate
+   - k8s {feature x} v1.1.1 container from remote image should run
+   - **Expect:** `current_owner=kube`, `container_version=1.1.1`, `remote_state=running`
+2. Set {feature x} desired owner to local (`set_owner=local`)
+   - k8s {feature x} v1.1.1 container from remote image should terminate
+   - local {feature x} v1.0.0 container should run
+   - **Expect:** `current_owner=local`, `container_version=1.0.0`, `remote_state=none`
+3. Set {feature x} desired owner to kube (`set_owner=kube`)
+   - local {feature x} v1.0.0 container should terminate
+   - k8s {feature x} v1.1.1 container from remote image should run
+   - **Expect:** `current_owner=kube`, `container_version=1.1.1`, `remote_state=running`
+
+### TC_LOCAL_KUBE_2: Upgrade Kube Feature v1.1.1 to Kube Feature v2.0.0 via Successful Manifest Application
+#### Test Objective
+Verify Device Under Test (DUT) running kube mode feature container v1.1.1 properly upgrades to kube mode feature v2.0.0 upon successful application of v2.0.0 manifest.
+#### Test Configuration
+- Pick up from step 3 of [TC_LOCAL_KUBE_1](#tc_local_kube_1-switch-between-local-mode-and-kube-mode)
+- {feature x} `current_owner=kube`, `set_owner=kube`, `remote_state=running`, `container_version=1.1.1`
```

Comment:
According to [Kubernetes support](https://github.com/renukamanavalan/SONiC/blob/kube_systemd/doc/kubernetes/Kubernetes-support.md), only `set_owner=kube` is configuration. Other items are status of DUT or feature. Probably it is better to explicitly specify what is required configuration and what is required status as precondition of a test.

Reply:
Thanks for the feedback, I updated the document to distinguish between desired status and configuration in Test Setup.